### ISSUE TITLE
XEP-0368: Test only checks the legacy port

### ISSUE THREAD

**Issue by mkoppmann** [iNPUTmice/caas#54]
The compliance test for “XEP-0368: SRV records for XMPP over TLS” only checks the deprecated 5223 port with `xmpps` and fails if just `_xmpp-client._tcp.example.net.` and `_xmpp-server._tcp.example.net.` are set.

If the server is running ejabberd, the recommendation links to [this](https://wiki.xmpp.org/web/SRV_Records) XMPP wiki article, which doesn’t mention the `xmpps` entries at all. In case of Prosŏdy, a recommendation with `xmpps` entries is listed. I had to look into the source to see that `xmpps` is required to pass this check.

**Comment by iNPUTmice** [comment]
The XEP-0368 checks for the xmpps records defined in that XEP. It is not hard coded to check for 5223; It will check what ever those records point to.

**Comment by mkoppmann** [comment]
I should have chosen a better title, that’s true. I also just realized, that the linked XMPP wiki article isn’t related to the XEP at all but just explains SRV records in general. So maybe the problem here is just that the ejabberd recommendation should be changed to link to the actual XEP so people can see, that they have to add the `xmpss` entries.

**Comment by lxp** [comment]
I was wondering about the same thing, as I had the opinion that direct TLS was deprecated long time ago. However, it seems to have gained popularity again :)

I think XEP-0368 is not really clear about if servers MUST provide direct TLS connections or not. It just specifies how the SRV records should look like, if the server supports it and how clients should handle the records and setup TLS connections to the server.
However, XEP-0387 clearly states that servers MUST accept direct TLS connections (see https://xmpp.org/extensions/xep-0387.html#nt-idm46142097138560).
So my conclusion is that the check is correct, but the result presentation and recommendation are a bit misleading.

There is one other thing that comes to my mind after reading XEP-0387 and XEP-0368.
Shouldn't this test also check for `_xmpps-server._tcp.example.net.` records?
I think the reason for not checking it, is that XEP-0387 does not require federation.
However, if a server has an `_xmpp-server._tcp.example.net.` record, it wants to federate.
I think XEP-0387 also applies to S2S connections, therefore, it should also have an `_xmpps-server._tcp.example.net.` record.

**Comment by moppman** [comment]
Here's something I don't get: XEP-0368 doesn't mention any ports specifically, but talks about 
> restrictive firewalls that only allow HTTPS.

Shouldn't this test then probe explicitly for port 443? The way it currently works is caas just checks if a connection on any of the ports defined in the server's dns svr entry can be established. This doesn't help people behind a hotel firewall, though. Or am I missing the point?

**Comment by licaon-kter** [comment]
@moppman That's up to the admin that set that port to 443 or whatever. Not sure why you'd probe 443 if the SRV doesn't point to it.

**Comment by moppman** [comment]
Yeah, that's what I was implying. Imho it should be expected from the admin to point the xmpps-client SVR record to 443, and this test should reflect that.

01.02.2020 21:24:50 Licaon_Kter <notifications@github.com>:
